docs/_includes/Types.md

```diff
@@ -974,6 +975,99 @@ if K in cache {  // check if temperature is in domain of cache
 }
 ```
 
+## 10.5. Iterating over collections
+
+Collections are very commonly used in programming and one frequently
+needs to iterate over the elements of a collection. Dafny does not have
+built-in iterator methods, but the idioms by which to do so are straightforward.
+The subsections below give some introductory examples; more
+detail can be found in this [power user note](http://leino.science/papers/krml275.html).
+
+TODO: Add examples of using a iterator class
+TODO: Should a foreach statment be added to Dafny
+
+### 10.5.1. Sequences and arrays
+
+Sequences and arrays are indexable and have a length. So the idiom to
+iterate over the contents is well-known. For an array:
+```dafny
+  var i: int := 0;
+  var sum: int := 0;
+  while i < s.Length
+    decreases s.Length - i
```

Comment:
This `decreases` clause is provided automatically, so I think it's better to omit it here.

Reply:
OK